ansible/configs/ocp4-workshop/lifecycle.yml

```diff
@@ -64,10 +64,11 @@
       when: ACTION == 'stop'
       ec2_instance:
         state: stopped
-        wait: no
+        wait: "{{ aws_instance_wait_for_stop }}"
         filters:
           "tag:guid": "{{ guid }}"
           "tag:env_type": "{{ env_type }}"
+          instance-state-name: running
```

Comment:
Why do we have to specify the state here ?

is there any other type that should be included ?  "starting"  "pending"  ... ?

I'd rather not touch this part, especially since i think this should be taken care of by the ansible module:

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_instance.py#L1532-L1542

Reply:
We were running into problems with trying to stop things that were not running like terminated instances.

Reply:
Did you open an issue to ansible/ansible project ?

it seems to me that this should be taken care by the module:

https://github.com/ansible/ansible/blob/devel/lib/ansible/modules/cloud/amazon/ec2_instance.py#L1533

It doesn't try to stop an already stopped instance, so it should not try to stop an instance marked  as Terminated.

The problem with the filter approach is that it's not exhaustive.

We can add more filters to reduce the set, but we can't do, as far i know, inclusive OR.

```
filters: running | pending | starting | ...
state: stopped
```